File: src/test/java/io/atesfactory/evrl/SpringIntegrationTests.java
package io.atesfactory.evrl;

import io.atesfactory.evrl.configuration.EvrlSpringContext;
import io.atesfactory.evrl.configuration.ProtocolResolver;
import io.atesfactory.evrl.loader.EvrlResourceLoader;
import org.junit.jupiter.api.Test;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.boot.test.context.SpringBootTest;
import org.springframework.context.ApplicationContext;
import org.springframework.core.env.Environment;
import org.springframework.core.io.DefaultResourceLoader;

import static org.junit.jupiter.api.Assertions.assertEquals;
import static org.junit.jupiter.api.Assertions.assertTrue;

@SpringBootTest
class SpringIntegrationTests {

    @Autowired
    private ApplicationContext applicationContext;

    @Autowired
    private Environment environment;

    @Test
    void shouldSetProtocolResolverToDefaultResourceLoader() {
        ProtocolResolver protocolResolver = new ProtocolResolver(
                new EvrlSpringContext(
                        applicationContext,
                        environment
                )
        );

        DefaultResourceLoader resourceLoader = new DefaultResourceLoader();
        protocolResolver.setResourceLoader(resourceLoader);

        assertTrue(resourceLoader.getProtocolResolvers().contains(protocolResolver));
    }

    @Test
    void shouldHaveDelegateClassLoader() {
        EvrlSpringContext evrlSpringContext =
                new EvrlSpringContext(applicationContext, environment);
        DefaultResourceLoader delegate = new DefaultResourceLoader();
        EvrlResourceLoader resourceLoader =
                new EvrlResourceLoader(evrlSpringContext, delegate);

        assertEquals(delegate.getClassLoader(), resourceLoader.getClassLoader());
    }
}

File: src/test/kotlin/io/atesfactory/evrl/SpringIntegrationTests.kt
package io.atesfactory.evrl

import io.atesfactory.evrl.configuration.ProtocolResolver
import io.atesfactory.evrl.configuration.EvrlSpringContext
import io.atesfactory.evrl.loader.EvrlResourceLoader
import org.junit.jupiter.api.Assertions.assertEquals
import org.junit.jupiter.api.Assertions.assertTrue
import org.junit.jupiter.api.Test
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.boot.test.context.SpringBootTest
import org.springframework.context.ApplicationContext
import org.springframework.core.env.Environment
import org.springframework.core.io.DefaultResourceLoader

@SpringBootTest
class SpringIntegrationTests {

    @Autowired
    lateinit var applicationContext: ApplicationContext

    @Autowired
    lateinit var environment: Environment

    @Test
    fun `should set protocol resolver to default resource loader`() {
        val protocolResolver = ProtocolResolver(
            EvrlSpringContext(
                applicationContext,
                environment
            )
        )

        val resourceLoader = DefaultResourceLoader()
        protocolResolver.setResourceLoader(resourceLoader)

        assertTrue(resourceLoader.protocolResolvers.contains(protocolResolver))
    }

    @Test
    fun `should have delegate class loader`() {
        val evrlSpringContext =
            EvrlSpringContext(applicationContext, environment)
        val delegate = DefaultResourceLoader()
        val resourceLoader =
            EvrlResourceLoader(evrlSpringContext, delegate)

        assertEquals(delegate.classLoader, resourceLoader.classLoader)
    }

}
